accounts-db/src/accounts_db.rs

```diff
@@ -6295,33 +6295,42 @@ impl AccountsDb {
         });
 
         // Always flush up to `requested_flush_root`, which is necessary for things like snapshotting.
-        let cached_roots: BTreeSet<Slot> = self.accounts_cache.clear_roots(requested_flush_root);
+        let flushed_roots: BTreeSet<Slot> = self.accounts_cache.clear_roots(requested_flush_root);
 
         // Iterate from highest to lowest so that we don't need to flush earlier
         // outdated updates in earlier roots
         let mut num_roots_flushed = 0;
         let mut flush_stats = FlushStats::default();
-        for &root in cached_roots.iter().rev() {
+        for &root in flushed_roots.iter().rev() {
             if let Some(stats) =
                 self.flush_slot_cache_with_clean(root, should_flush_f.as_mut(), max_clean_root)
             {
                 num_roots_flushed += 1;
                 flush_stats.accumulate(&stats);
             }
 
-            // Regardless of whether this slot was *just* flushed from the cache by the above
-            // `flush_slot_cache()`, we should update the `max_flush_root`.
-            // This is because some rooted slots may be flushed to storage *before* they are marked as root.
-            // This can occur for instance when
-            //  the cache is overwhelmed, we flushed some yet to be rooted frozen slots
-            // These slots may then *later* be marked as root, so we still need to handle updating the
-            // `max_flush_root` in the accounts cache.
+            if let Some(storage) = self.storage.get_slot_storage_entry(root) {
```

Comment:
https://github.com/anza-xyz/agave/blob/8db563d3bba4d03edf0eb2737fba87f394c32b64/accounts-db/src/accounts_db.rs#L7564

These slots has already been added to uncleaned_pubkeys when calculating accounts_delta_hash. 

When clean runs, it will include all these dirty pubkeys in the candidates. 

https://github.com/anza-xyz/agave/blob/8db563d3bba4d03edf0eb2737fba87f394c32b64/accounts-db/src/accounts_db.rs#L2477

Therefore, I think we don't need to add them to dirty store at flush. It seems redundant and is a waste of time to scan those stores again.

Reply:
I updated the code to only added to dirty_stores when the slot is not found in uncleaned_pubkeys, and a stat to count how many time we added them. 
I think it should be zero. I have started a node the verify this. We will know soon.

Reply:
Yes, I think we can do that.

Reply:
The experimental run on mainnet shows that the stats are indeed zeros. This means that we don't need to add them to dirty storage at flush time. 

I will shelve this PR, and start two new other PRs for the obsolete comment change and the fetch_max optimization change.